src/dirs.rs

```diff
@@ -12,18 +14,39 @@ pub fn is_absolute_path<S: Into<std::ffi::OsString>>(path: S) -> Option<PathBuf>
     }
 }
 
-pub fn home_dir() -> Option<PathBuf> {
-    env::var_os("HOME").map(PathBuf::from)
+pub fn has_home_dir() -> bool {
+    env::var_os("HOME").is_some()
 }
 
-pub fn config_dir() -> Option<PathBuf> {
+pub fn home_dir() -> PathBuf {
+    PathBuf::from(env::var_os("HOME").unwrap())
+}
+
+pub fn config_dir() -> PathBuf {
     env::var_os("XDG_CONFIG_HOME")
```

Comment:
I read only `XDG_RUNTIME_DIR` is expected to be there and the others should be set by the user. We could default to cwd if we can't find any of the `XDG*` directories?

Reply:
I guess it is fine as we have meaningful defaults when we can't find any of them. Ignore my comment.

Reply:
[spec](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)